arealite/README.md

```diff
@@ -0,0 +1,213 @@
+
+#### 3. **Configuration-Driven Architecture**
+All components are configured through dataclasses defined in `cli_args.py`, enabling:
+- Type-safe configuration
+- Easy CLI argument generation
+- Clear documentation of available options
+
+
+## Implementation Details
+
+### Training Pipeline
+
+1. **Initialization**: Factory classes create configured instances of engines, trainers, and rollout collectors
+2. **Rollout Phase**: `RolloutController` coordinates async data collection across multiple `RolloutWorker` instances
+3. **Training Phase**: `Trainer` performs synchronous gradient updates using collected data
+4. **Weight Updates**: Updated model weights are pushed to LLM servers via `update_weights_to()`
+
+### Rollout System
+
+The rollout system supports arbitrary agentic rollout paradigms, implemented as `RolloutCollector` instances. `RolloutCollector` exposes a `run_episode` method for users to implement the logic of collecting a complete agentic trajectory. Users can implement gymnasium-compatible `Agent` and `Environment` interfaces first and combine them as a collector as in normal RL literature (in `arealite/impl/agentic/`), or users can implement the collector directly if the agent-environment interfaces are not compatible with the desired use cases (in `arealite/impl/rlvr/`).
+
+## Expected Usage
+
+### Basic RL Training
+```bash
+python3 arealite.cli.main \
+    experiment_name=my-exp trial_name=my-trial \
+    trainer.type=grpo \
+    trainer.grpo.actor.path=Qwen/Qwen2-0.5B
+```
+
+### Rollout-Only Evaluation
+```bash
+python3 arealite.cli.main \
+    trainer.type=null \
+    valid_dataset.path=huggingface/dataset
+```
+
+### Distributed Training
+```bash
+python3 arealite.cli.main \
+    mode=ray \
+    allocation_mode=sglang.d16p1m1+d32p2m1 \
+    trainer.type=grpo
+```
+
+## Customization Guide
+
+### Adding New Trainers
+
+1. **Implement trainer class** in `impl/trainer/`:
+```python
+from arealite.api.trainer_api import Trainer
+
+class MyTrainer(Trainer):
+    def train(self, resume_from_checkpoint=None):
+        # Implementation here
+        pass
+```
+
+2. **Add configuration** in `cli_args.py`:
+```python
+@dataclass  
+class MyTrainerConfig:
+    learning_rate: float = 1e-4
+```
+
+3. **Register in factory** in `trainer_api.py`:
+```python
+def make_trainer(self, config: TrainerConfig) -> Trainer:
+    if config.type == "my_trainer":
+        return MyTrainer(...)
+```
+
+### Adding New Rollout Collectors
+
+1. **Implement collector** in `impl/`:
+```python
+from arealite.api.rollout_api import RolloutCollector
+
+class MyCollector(RolloutCollector):
+    async def arun_episode(self, gconfig, env_option=None, seed=None):
+        # Implementation here
+        pass
+```
+
+2. **Register in factory** in `rollout_api.py`:
+```python
+def make_collector(self, config: RolloutCollectorConfig):
+    if config.type == "my_collector":
+        return MyCollector(...)
+```
+
```

Comment:
One thing in production system often lacks is systematic support of reward model, verifier and generative reward model. From the RL point of view it can be encapsulated in the concept of environment but it is curbersome to integrate them. Method a bit here might be good for those who are using RL in production.